src/Widgets/SetupWidget.ts

```diff
@@ -0,0 +1,51 @@
+
+import { Intent } from "matrix-bot-sdk";
+import { BridgeWidgetConfig } from "../Config/Config";
+import LogWrapper from "../LogWrapper";
+import { HookshotWidgetKind } from "./WidgetKind";
+const log = new LogWrapper("SetupWidget");
+
+export class SetupWidget {
+    static async SetupRoomConfigWidget(roomId: string, botIntent: Intent, config: BridgeWidgetConfig): Promise<boolean> {
+
+        const widgetTitle = config.branding?.widgetTitle || "Hookshot Configuration";
+
+        const widgetKey = "hookshot_room_config";
+        log.info(`Running setupWidget for ${roomId}`);
```

Comment:
This looks very similar to what AdminRoom is doing in `setupWidget()`, maybe it's worth reusing some of it.